### PR TITLE
Add support for more major modes

### PR DESCRIPTION
**Pull request by jscheid**

### CODE REVIEW

File: test-cases/svelte-mode/a.original.svelte [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<script>
```
Comment (asbish) [inline test-cases/svelte-mode/a.original.svelte:1]: These three lines are the same as `a.prettier.svelte`. How about this?
```
<script >
   let  name  =  "world"
 </script>
```

**Comment by asbish** [comment]
Awesome work!

I have one question about these version strings. https://github.com/jscheid/prettier.el/blob/b315e2ed721c46f6b42d3d543e6d438536469b2c/prettier.el#L172
When do we need to update the strings in accordance with the prettier version in each plugin's `dependencies` or `peer-dependencies`?

File: test-cases/svelte-mode/a.original.svelte (hunk 1 above)
Comment (jscheid) [reply test-cases/svelte-mode/a.original.svelte:1]: Good idea, done.

**Comment by jscheid** [comment]
> I have one question about these version strings.
> 
> https://github.com/jscheid/prettier.el/blob/b315e2ed721c46f6b42d3d543e6d438536469b2c/prettier.el#L172
> 
> When do we need to update the strings in accordance with the prettier version in each plugin's `dependencies` or `peer-dependencies`?

Good question. The one you quoted (Angular) is actually a core plugin that [was added to Prettier in version 1.15](https://prettier.io/blog/2018/11/07/1.15.0.html#support-html-vue-and-angular-5259-by-ikatyang-4753-by-evilebottnawi-2083-by-azz), so in this case the version information won't change (unless they remove support one day). But your question still holds for external plugins, one could drop support for an older Prettier version at any time.

Then again, people are still free to use an older version of the plugin, so technically the minimum Prettier version won't change here either.

In any case, even if some of the version information (which I thought we'd provide as a courtesy) was flat out wrong, I don't think much harm will be done. Worst case, if someone gets bitten by an inaccuracy they can point it out and we'd update it. Would you agree?

If you think it better to remove the version info altogether and let people work it out themselves, I wouldn't be opposed to that either.

**Comment by jscheid** [comment]
On second thought, perhaps it would be good to drop the version info for external plugins and keep it for internal ones, since for those it is essential information: the only way to get support for an input type such as Angular is to upgrade to a Prettier version that supports it.

**Comment by asbish** [comment]
Thank you for your detailed explanations. I agree with your first thought.
If we get issues or some versions regarding external plugins become uncertain, it would be better to remove them in accordance with your second thought.
Thanks again!

**Pull request merged**